webdriver/tests/interaction/element_click.py

```diff
@@ -0,0 +1,214 @@
+import pytest
+
+from tests.support.asserts import assert_error, assert_success
+from tests.support.inline import inline
+
+def click(session, element):
+    return session.transport.send("POST", "session/{session_id}/element/{element_id}/click"
+                                  .format(session_id=session.session_id,
+                                          element_id=element.id))
+
+
+# 14.1 Element Click
+
+def test_no_browsing_context(session, create_window):
+    # 14.1 step 1
+    session.url = inline("<p>This is not an editable paragraph.")
+    element = session.find.css("p", all=False)
+
+    session.window_handle = create_window()
+    session.close()
+
+    response = click(session, element)
+    assert_error(response, "no such window")
+
+
+def test_element_file_upload_state(session):
+    # 14.1 Step 3
+    session.url = inline("<input type=file value=Federer>")
+
+    element = session.find.css("input", all=False)
+    response = click(session, element)
+    assert_error(response, "invalid argument")
+
+
+def test_scroll_into_element_view(session):
+    # 14.1 Step 4
+    session.url = inline("<input type=text value=Federer><div style=\"height: 200vh; width: 5000vh\">")
+
+    # Scroll to the bottom right of the page
+    session.execute_script("window.scrollTo(document.body.scrollWidth, document.body.scrollHeight);")
+    element = session.find.css("input", all=False)
+    # Clear and scroll back to the top of the page
+    response = click(session, element)
+    assert_success(response)
+
+    # Check if element cleared is scrolled into view
+    response = session.execute_script("""rect = document.getElementsByTagName("input")[0].getBoundingClientRect();
+                                        xOffset = window.pageXOffset;
+                                        yOffset = window.pageYOffset;
+                                        return  rect["top"] < (window.innerHeight + yOffset) &&
+                                                rect["left"] < (window.innerWidth + xOffset) &&
+                                                (rect["top"] + rect["height"]) > yOffset &&
+                                                (rect["left"] + rect["width"]) > xOffset;""")
+    assert response is True
+
+
+@pytest.mark.parametrize("element", ["""<div style="width: 500px; height: 100px; background-color: green; transform: translate(-50px, -50px);">
+                                            <a href=#>aaaa</a>
+                                        </div>""",
+                                    """<div style="width: 500px; height: 100px; background-color: green; transform: rotate(50deg);">
+                                            <a href=#>aaaa</a>
+                                        </div>"""])
```

Comment:
This is not an appropriate use of parametrization in my opinion. Can you put the HTML in the test and parameterize the transform value instead?

Reply:
fixed